### ISSUE TITLE
Counterintuitive bids.File interface

### ISSUE THREAD

**Issue by nbeliy** [bids-standard/bids-matlab#282]
I find the new interface to `bids.File`, trying to mix structure and class fields.
For example function `create_filename` don't create a file name, but makes a copy of initial object with some changed fields.
As using objects, clone objects is trivial, I propose to instead of functions `create_filename`, `create_rel_path` etc. use one function `update()` that based on values of internal fields changes filename, relative path etc.

In that way, user can parse a file, change fields directly and regenerate new name in transparent way:
```matlab
file = bids.File('sub-AD0041_desc-sum_rangeLow-9_rangeHigh-22_pet.nii');
file.prefix = 'abcde';
file.update();
file.filename

  ans = 'abcdesub-AD0041_desc-sum_rangeLow-9_rangeHigh-22_pet.nii'
```
Should I make a merge request for that?

**Comment by Remi-Gau** [comment]
Yes that you would make more sense indeed. Open a PR and we will take it from there.

**Comment by Remi-Gau** [comment]
closed by #289
